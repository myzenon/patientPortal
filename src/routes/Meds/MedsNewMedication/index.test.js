import React from 'react'
import { MedsNewMedication } from './index'
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

test('renders correctly', () => {
    const props = {
        cmr: { ...mockObj.cmr,
          otherMedication_1: {
              display: "ASPIRIN 200",
              brandName: "ASPIRIN",
              genericName: "ASPIRIN",
              dosageStrength: "200",
              taken: "y",
              reasonCode: [
                  { conditionName: 'Other', description: ' ' }
              ],
              dosage:{ description: "ASPIRIN"},
              hasQuantity: true
          } },
        match: {
            params: {
                medId: "otherMedication_1"
            }
        },
    }
    console.log(props.cmr)
    const wrapper = render(
        <MedsNewMedication { ...props } />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})

test.skip('NEXT button calls /fhir/cmr API to save this Other-Medication | goes to the [ENTER NEW MEDICATION DOSING] page', done => {
    const props = {
      cmr: { ...mockObj.cmr,
        otherMedication_1: {
            display: "ASPIRIN 200",
            brandName: "ASPIRIN",
            genericName: "ASPIRIN",
            dosageStrength: "200",
            taken: "y",
            hasQuantity: true
        } },
        match: {
            params: {
                medId: "otherMedication_1"
            }
        },
        callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirCmrIdPut')
            expect(get).toEqual({
                id: mockObj.cmr.id
            })
            expect(post).toEqual({
                otherMedication_1: {
                    display: "ASPIRIN 200",
                    brandName: "ASPIRIN",
                    genericName: "ASPIRIN",
                    dosageStrength: "200",
                    taken: "y",
                    hasQuantity: true
                }
            })
            return Promise.resolve()
        },
        setCMR: (cmr) => {
            expect(cmr).toEqual({
                ...mockObj.cmr,
                otherMedication_1: {
                    display: "ASPIRIN 200",
                    brandName: "ASPIRIN",
                    genericName: "ASPIRIN",
                    dosageStrength: "200",
                    taken: "y",
                    hasQuantity: true
                }
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/meds-other-whatfor/otherMedication_1')
            done()
        }
    }
    const wrapper = mount(
        <MedsNewMedication { ...props } />
    )
    wrapper.find('input[name="medicine"]').simulate('change', {
        target: {
            name: 'medicine',
            value: 'aspirin'
        }
    })
    wrapper.find('input[name="dose"]').simulate('change', {
        target: {
            name: 'dose',
            value: '200'
        }
    })
    wrapper.find('input[name="description"]').simulate('change', {
        target: {
            name: 'dosage',
            value: 'take 2'
        }
    })
    wrapper.find('button[type="submit"]').simulate('submit')
})
