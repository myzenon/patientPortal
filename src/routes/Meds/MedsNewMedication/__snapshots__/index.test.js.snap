// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <br />
     
    <p />
    <form
      class="formWrapper"
    >
      <div
        class="row"
      >
        <span
          class="questions"
        >
          Please tell us about your over-the-counter medication or herbal supplement and how you take it.
        </span>
        <p />
      </div>
      <div
        class="row"
      >
        <div
          class="formLabel col-6"
        >
          <span
            class="inputLabel"
          >
            Medicine Name
          </span>
          <div
            class="inputTextWrapper"
          >
            <input
              autofocus=""
              class="inputTextCenter"
              name="medicine"
              placeholder="For example: Advil"
              textalign="center"
              type="text"
              value="ASPIRIN"
            />
          </div>
        </div>
        <div
          class="formLabel col-6"
        >
          <span
            class="inputLabel"
          >
            Dosage Strength
          </span>
          <div
            class="textWrapper"
          >
            <input
              class="inputTextCenter"
              name="dose"
              placeholder="For example: 200mg"
              textalign="center"
              type="text"
              value="200"
            />
          </div>
        </div>
      </div>
      <br />
      <br />
      <div
        class="row"
      >
        <div
          class="formLabel col"
        >
          <span
            class="inputLabel"
          >
            Describe how you take this medicine
          </span>
          <div
            class="inputWrapper"
          >
            <input
              class="inputTextCenter"
              name="dosage"
              placeholder="I take 1 to 2 tablets every 4 to 6 hours as needed"
              textalign="center"
              type="text"
              value="ASPIRIN"
            />
          </div>
        </div>
      </div>
      <br />
      <p>
        <button
          class="button btn btn-secondary"
          type="submit"
        >
          <b>
              NEXT 
          </b>
        </button>
      </p>
    </form>
  </div>
  <div
    class="footerWrapper container"
  >
    <div>
      <div
        class="wrapper"
      >
        <div
          class="stepCompleted"
        >
          Account
          <br />
          Setup
          <div
            class="stepNumberActive"
          >
            1
          </div>
        </div>
        <div
          class="stepCompleted"
        >
          Medication
          <br />
          Review
          <div
            class="stepNumberActive"
          >
            2
          </div>
        </div>
        <div
          class="stepActive"
        >
          Additional 
          <br />
          Medications
          <div
            class="stepNumberActive"
          >
            3
          </div>
        </div>
        <div
          class="step"
        >
          Health
          <br />
          Updates
          <div
            class="stepNumber"
          >
            4
          </div>
        </div>
        <div
          class="step"
        >
          Set
          <br />
          Appointment
          <div
            class="stepNumber"
          >
            5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
